### PR TITLE
Sa/seifert beheng2006

### PR DESCRIPTION
**Pull request by sajjadazimi**
## Purpose 
Add 2M parametrization of Seifert and Beheng (2006). We already have SF2001 implemented. This PR implements SF2006 which is based on and expands SF2001. 2M parametrizations for raindrops breakup, terminal velocity and rain evaporation are added.

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.
- I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/Microphysics2M.jl [hunk 1]
```diff
@@ -219,15 +293,170 @@ function rain_self_collection(
...
+    Dr_th::FT = CMP.Dr_th_br_SB2006(param_set)
+    kbr::FT = CMP.kbr_SB2006(param_set)
+    κbr::FT = CMP.κbr_SB2006(param_set)
+
+    L_rai = ρ * q_rai
+    xr = mean_mass_rai_limited(param_set, L_rai, N_rai)
+    Dr = (xr * 6 / π / ρw)^FT(1 / 3)
+    ΔD = Dr - Deq
+    phi_br =
+        (Dr < Dr_th) ? FT(-1) : ((ΔD <= 0) ? kbr * ΔD : 2 * (exp(κbr * ΔD) - 1))
```
Comment (trontrytel) [inline src/Microphysics2M.jl:349]: I saw your note in the docs, but to me it looks like it's still not contuntous?

In the linear regime I have `1000 * 0.9 * 1e-3` vs in the exponential regime `2 * (exp(2.3 * 1e3 * 0.9 * 1e-3) -1) = 13.849646235698973`?

**Review by trontrytel — APPROVED** [review]
Thank you for adding this scheme. This is great. I left some comments and questions. 

Would be great to also add more plots to the documentation that show how individual process rates behave as a function of N and q. But we can do it later, maybe during the SURF project this summer.

File: src/Microphysics2M.jl (hunk 1 above)
Comment (sajjadazimi) [reply src/Microphysics2M.jl:349]: > I saw your note in the docs, but to me it looks like it's still not contuntous?
> 
> In the linear regime I have `1000 * 0.9 * 1e-3` vs in the exponential regime `2 * (exp(2.3 * 1e3 * 0.9 * 1e-3) -1) = 13.849646235698973`?

In both linear and exponential functions, the input is Delta D which is D - Deq. So in the linear regime `1000 * (0.9 -0.9) =0` equal `2*(exp(2.3 * 1e3 * (0.9 - 0.9)) - 1) = 0` in the exponential regime.

File: src/Microphysics2M.jl [hunk 2]
```diff
@@ -78,24 +146,27 @@ function autoconversion(
...
+    kcc::FT = CMP.kcc_SB2006(param_set)
+    νc::FT = CMP.νc_SB2006(param_set)
+    x_star::FT = CMP.xr_min_SB2006(param_set)
+    ρ0::FT = CMP.ρ0_SB2006(param_set)
+    A::FT = CMP.A_phi_au_SB2006(param_set)
+    a::FT = CMP.a_phi_au_SB2006(param_set)
+    b::FT = CMP.b_phi_au_SB2006(param_set)
 
     L_liq = ρ * q_liq
-    x_liq = L_liq / N_liq
```
[the remark this answers is not in the document]
Comment (sajjadazimi) [reply src/Microphysics2M.jl:89]: This is an interesting consideration. This approximation is good as long as variations of dry-air density with time are negligible: `dL/dt = rho dq/dt + q drho/dt \approx rho dq/dt + q d rho_dry/dt + q rho_dry d q_tot / dt`.

File: src/Microphysics2M.jl [hunk 3]
```diff
@@ -219,15 +293,170 @@ function rain_self_collection(
...
+    krr::FT = CMP.krr_SB2006(param_set)
+    κrr::FT = CMP.κrr_SB2006(param_set)
+    d::FT = CMP.d_sc_SB2006(param_set)
+    ρw::FT = CMP.ρ_cloud_liq(param_set)
+    ρ0::FT = CMP.ρ0_SB2006(param_set)
 
     L_rai = ρ * q_rai
+    λr =
+        lambda_rai_limited(param_set, L_rai, N_rai) *
+        (SF.gamma(4) / FT(π) / ρw)^FT(1 / 3)
```
[the remark this answers is not in the document]
Comment (sajjadazimi) [reply src/Microphysics2M.jl:305]: > Why do we have to multiply lambda by `(SF.gamma(4) / FT(π) / ρw)^FT(1 / 3)` ?

Here in the function for self-collection, lambda is the parameter of the gamma distribution based on drops mass, while the function `lambda_rai_limited` returns lambda for the distribution based on drops diameter. So this function return must be converted by this simple multiplication to the fixed number.

**Pull request merged**